File: LeetCode - Algorithm/Two Pointers/392.js
// 392. Is Subsequence
// https://leetcode.com/problems/is-subsequence/

/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  left = 0;
  right = 0;

  while (left <= right && right < nums.length) {
    console.log(left, right, nums);

    if (nums[left] === 0 && nums[right] !== 0) {
      temp = nums[left];
      nums[left] = nums[right];
      nums[right] = temp;

      left += 1;
      right += 1;
    } else if (nums[left] === 0 && nums[right] === 0) {
      right += 1;
    } else if (nums[left] !== 0 && nums[right] !== 0) {
      left += 1;
    } else {
      left += 1;
      right += 1;
    }
  }
};

File: LeetCode - Algorithm/Two Pointers/392.py
# 392. Is Subsequence
# https://leetcode.com/problems/is-subsequence/


class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        first = 0
        second = 0

        while first < len(s) and second < len(t):
            if s[first] == t[second]:
                first += 1
                second += 1
            else:
                second += 1

        if first >= len(s):
            return True
        else:
            return False
